### DISCUSSION TITLE
How do I find out the Gateway's encryption settings?

### DISCUSSION THREAD

**Discussion by tjmcclure0501** [strongswan/strongswan#261] [Q&A]
https://support.keriocontrol.gfi.com/hc/en-us/articles/360015179360-Configuring-IPSec-VPN-Client-on-Linux-Debian-Based-OS

My conf file:
conn %default
    ikelifetime=60m
    keylife=20m
    rekeymargin=3m
    keyingtries=1
    keyexchange=ikev1
    authby=secret

conn VPN1
    keyexchange=ikev1
    left=%defaultroute
    auto=add
    ike="aes128-sha1-modp2048,3des-sha1-modp1536"
    esp="aes128-sha1,3des-sha1"
    authby=secret
    type=transport
    leftprotoport=17/1701
    rightprotoport=17/1701
    right=50.213.203.33

I used the link above to install and configure strongswan and it seems to be successful.  the service strongswan was installed.   I try to connect I get the following error:
initiating Main Mode IKE_SA VPN1[4] to 50.213.203.33
generating ID_PROT request 0 [ SA V V V V V ]
sending packet: from 10.0.0.206[500] to 50.213.203.33[500] (272 bytes)
received packet: from 50.213.203.33[500] to 10.0.0.206[500] (188 bytes)
parsed INFORMATIONAL_V1 request 0 [ N(NO_PROP) ]
received NO_PROPOSAL_CHOSEN error notify
establishing connection 'VPN1' failed

I suspect that the gateway is using an encryption different from what I am proposing.  I set the log level to the highest:
sudo  ipsec stroke loglevel ike 4

I then viewed the  log for the service:
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 06[CFG] received stroke: loglevel 4 for ike
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 08[CFG] received stroke: initiate 'VPN1'
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] queueing ISAKMP_VENDOR task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] queueing ISAKMP_CERT_PRE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] queueing MAIN_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] queueing ISAKMP_CERT_POST task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] queueing ISAKMP_NATD task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] queueing QUICK_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_VENDOR task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] activating new tasks
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_CERT_PRE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_CERT_PRE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing MAIN_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing MAIN_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_CERT_POST task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_CERT_POST task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_NATD task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_NATD task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing QUICK_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing QUICK_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] activating new tasks
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] activating new tasks
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_VENDOR task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_VENDOR task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_CERT_PRE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_CERT_PRE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating MAIN_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating MAIN_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_CERT_POST task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_CERT_POST task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_NATD task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 09[ENC] parsed INFORMATIONAL_V1 request 0 [ N(NO_PROP) ]
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 09[IKE] received NO_PROPOSAL_CHOSEN error notify
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 09[IKE] IKE_SA VPN1[2] state change: CONNECTING => DESTROYING
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 06[CFG] received stroke: loglevel 4 for ike
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 08[CFG] received stroke: initiate 'VPN1'
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] queueing ISAKMP_VENDOR task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] queueing ISAKMP_CERT_PRE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] queueing MAIN_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] queueing ISAKMP_CERT_POST task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] queueing ISAKMP_NATD task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] queueing QUICK_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_VENDOR task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net ipsec[18656]: 10[IKE] activating new tasks
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_CERT_PRE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_CERT_PRE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing MAIN_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing MAIN_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_CERT_POST task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_CERT_POST task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_NATD task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing ISAKMP_NATD task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing QUICK_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] queueing QUICK_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] activating new tasks
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] activating new tasks
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_VENDOR task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_VENDOR task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_CERT_PRE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_CERT_PRE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating MAIN_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating MAIN_MODE task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_CERT_POST task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_CERT_POST task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_NATD task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE]   activating ISAKMP_NATD task
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] sending XAuth vendor ID
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] sending XAuth vendor ID
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] sending DPD vendor ID
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] sending DPD vendor ID
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] sending FRAGMENTATION vendor ID
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] sending FRAGMENTATION vendor ID
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] sending NAT-T (RFC 3947) vendor ID
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] sending NAT-T (RFC 3947) vendor ID
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] sending draft-ietf-ipsec-nat-t-ike-02\n vendor ID
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] sending draft-ietf-ipsec-nat-t-ike-02\n vendor ID
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] initiating Main Mode IKE_SA VPN1[3] to 50.213.203.33
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] initiating Main Mode IKE_SA VPN1[3] to 50.213.203.33
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] IKE_SA VPN1[3] state change: CREATED => CONNECTING
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[IKE] IKE_SA VPN1[3] state change: CREATED => CONNECTING
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[ENC] generating ID_PROT request 0 [ SA V V V V V ]
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 10[NET] sending packet: from 10.0.0.206[500] to 50.213.203.33[500] (272 bytes)
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 06[NET] received packet: from 50.213.203.33[500] to 10.0.0.206[500] (188 bytes)
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 06[ENC] parsed INFORMATIONAL_V1 request 0 [ N(NO_PROP) ]
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 06[IKE] received NO_PROPOSAL_CHOSEN error notify
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 06[IKE] received NO_PROPOSAL_CHOSEN error notify
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 06[IKE] IKE_SA VPN1[3] state change: CONNECTING => DESTROYING
Mar 15 12:07:32 5df3bfb7d42b5.streamlock.net charon[18679]: 06[IKE] IKE_SA VPN1[3] state change: CONNECTING => DESTROYING

How do I know what encryption the gateway is trying to use?  And do I need to change both of the following lines in the conf file:
    ike="aes128-sha1-modp2048,3des-sha1-modp1536"
    esp="aes128-sha1,3des-sha1"

**Comment by Thermi** [comment]
You either need to figure that out by reviewing logs of the other side, or make the other peer connect to you. Then you get to see what proposal it uses.


Btw, you should probably switch to either IKEv1 without L2TP, or to IKEv2. IKEv2 preferably. IKEv1 is terribly old.

EDIT: I just saw that you don't have logs from the other side at all. You should probably take a look at it, because NO_PROPOSAL_CHOSEN can also mean the other peer doesn't have any config for you at all.

**Comment by strongX509** [comment]
Using
```
sudo ipsec stroke loglevel cfg 2
```
you will see your own proposals  but as an initiator you never will be able to see the proposals supported by the peer side. 

You could just delete the `ike="aes128-sha1-modp2048,3des-sha1-modp1536"` line in you configuration and strongSwan would then send its default proposals. If you're lucky there will be one cipher suite among them acceptable by the peer side. I'm not sure if newer strongSwan versions propose "sha1" by default but you will see it if you increase the loglevel to cfg 2.
